# will be fixed in nextra 2.2.19
diff --git a/dist/loader.mjs b/dist/loader.mjs
index bc84edbfd2653368061731a6a4a2042c489ade07..fcaa22e702f55ece2c496a7ed1a55ebfcf51f3ac 100644
--- a/dist/loader.mjs
+++ b/dist/loader.mjs
@@ -175,8 +175,8 @@ async function loader(context, source) {
     headings,
     hasJsxInH1,
     timestamp,
+    pageMap,
     ...!HAS_UNDERSCORE_APP_MDX_FILE && {
-      pageMap,
       flexsearch,
       newNextLinkBehavior
     },
@@ -200,10 +200,13 @@ ${finalResult}
 
 const __nextra_internal__ = globalThis[Symbol.for('__nextra_internal__')] ||= Object.create(null)
 __nextra_internal__.Layout = __nextra_layout
-__nextra_internal__.pageMap = ${JSON.stringify(pageMap)}
+__nextra_internal__.pageMap = ${JSON.stringify(pageOpts.pageMap)}
 __nextra_internal__.flexsearch = ${JSON.stringify(flexsearch)}
 ${themeConfigImport ? "__nextra_internal__.themeConfig = __nextra_themeConfig" : ""}`;
   }
+  if (HAS_UNDERSCORE_APP_MDX_FILE) {
+    delete pageOpts.pageMap;
+  }
   const stringifiedPageOpts = JSON.stringify(pageOpts);
   const stringifiedChecksum = IS_PRODUCTION ? "''" : JSON.stringify(hashFnv32a(stringifiedPageOpts));
   const dynamicMetaModules = dynamicMetaItems.map(
# For some reasons I get build error for the following pages
#
# Error: Export encountered errors on following paths:
#         /ar/cookbook/cosmos
#         /es/cookbook/cosmos
#         /ja/cookbook/cosmos
#         /ko/cookbook/cosmos
#         /vi/cookbook/cosmos
#         /zh/cookbook/cosmos
#
# While visiting locally /ar/cookbook/cosmos
# useRouter() returs route which is /en/cookbook/cosmos

diff --git a/dist/use-internals.mjs b/dist/use-internals.mjs
index 43482104268d7086e219a0704497f2b3bbe8ddb2..ce08afb3cd60c99ddd3fc7b165faf8d32ebf0033 100644
--- a/dist/use-internals.mjs
+++ b/dist/use-internals.mjs
@@ -17,7 +17,7 @@ function useInternals() {
       };
     }, [route, __nextra_internal__.refreshListeners, rerender]);
   }
-  const context = __nextra_internal__.context[route];
+  const context = __nextra_internal__.context[__nextra_internal__.route]
   if (!context) {
     throw new Error(
       `No content found for the current route. This is a Nextra bug.`
